Mathlib/CategoryTheory/Galois/Basic.lean

```diff
@@ -202,4 +206,29 @@ lemma evaluation_aut_injective_of_connected (A : C) [ConnectedObject A] (a : F.o
 
 end PreGaloisCategory
 
+/-- A `PreGaloisCategory` is a `GaloisCategory` if it admits a fibre functor. -/
+class GaloisCategory (C : Type u₁) [Category.{u₂, u₁} C]
+    extends PreGaloisCategory C : Prop where
+  hasFibreFunctor : ∃ F : C ⥤ FintypeCat.{w}, Nonempty (PreGaloisCategory.FibreFunctor F)
```

Comment:
With this definition, `GaloisCategory C` depends on a universe `w` that is unrelated to those that are related to `C`. I think it would be advisable to replace `w` by `u₂` in this definition. (Hopefully, with this definition, the example of finite `G`-sets would be a `GaloisCategory`. Then, in future developments of the API, it could be shown that the existence of a fiber functor `C ⥤ FintypeCat.{w}` is independent of the universe `w`.)

Reply:
I suppose, we can't write there exists a universe `w` and a fibre functor `F : C ⥤ FintypeCat.{w}`? Finite `G`-sets is a `GaloisCategory` anyway:
```
/-- The category of finite `G`-sets is a `GaloisCategory`. -/
instance : GaloisCategory (Action FintypeCat (MonCat.of G)) where
  hasFibreFunctor := ⟨Action.forget FintypeCat (MonCat.of G), ⟨inferInstance⟩⟩
```
compiles. I will add it to the other PR, to avoid conflicts.

Reply:
Should all the `FintypeCat` categories be equivalent? One could add the lemma that a functor `F : C ⥤ FintypeCat.{w}` is a `FibreFunctor` if and only if it is so after composing with an equivalence to some other `FintypeCat.{w'}`.

Reply:
> I will add it to the other PR, to avoid conflicts.

I accidentally pushed it to this branch and now realized that on the other one, `GaloisCategory` is not defined yet. So I'll now keep the instance here and when this one is merged, merge master in the other one.

Reply:
> Shouldn't all the FintypeCat categories be equivalent?

Yes, of course, but this would require a proof, following the pattern you suggested. In this future API, we could introduce the fact that fibre functors behave well with pre/post-composition with an equivalence. (Only the `preservesQuotientsByFiniteGroups` should require some work as we would may have to change the universes of `G`.) We would presumably also a definition like `FibreFunctor.ofIso`: if `F` and `G` are isomorphic functors, and `F` is a fibre functor, then `G` also is.